classes/patreon_routing.php

```diff
@@ -94,33 +118,29 @@ function parse_request( &$wp ) {
 				if(array_key_exists('error', $tokens)) {
 
 					/* redirect to homepage because of some error #HANDLE_ERROR */
-					wp_redirect( home_url() );
+					$redirect = add_query_arg( 'patreon_message', 'patreon_cant_login_api_error', $redirect);
+					wp_redirect( $redirect );
 					exit;
 
 				} else {
 
-					$redirect = false;
-					if(get_option('patreon-enable-redirect-to-page-after-login', false)) {
-						$redirect = get_option('patreon-enable-redirect-to-page-id', get_option('page_on_front') );
-					} else if(isset($_COOKIE['ptrn_dst']) && is_numeric($_COOKIE['ptrn_dst'])) {
-						$redirect = get_post($_COOKIE['ptrn_dst']);
-						unset($_COOKIE['ptrn_dst']);
-					}
-
-					$redirect = apply_filters('ptrn/redirect', $redirect);
-
-					/* redirect to homepage successfully #HANDLE_SUCCESS */
 					$api_client = new Patreon_API($tokens['access_token']);
+					
 					$user_response = $api_client->fetch_user();
+			
+					if(apply_filters('ptrn/force_strict_oauth',get_option('patreon-enable-strict-oauth', false))) {
 
-					if(get_option('patreon-enable-strict-oauth', true)) {
-						$user = Patreon_Login::updateLoggedInUser($user_response, $tokens, $redirect);
+						$user = Patreon_Login::updateLoggedInUserForStrictoAuth($user_response, $tokens, $redirect);
 					} else {
-						$user = Patreon_Login::createUserFromPatreon($user_response, $tokens, $redirect);
+								
```

Comment:
remove this extra line

Reply:
Committed:

https://github.com/codebard/patreon-wordpress/commit/44c130ff6fc149bb2df226fb9c1383a2b74a63a8